### ISSUE TITLE
template-curly-spacing does not always work

### ISSUE THREAD

**Issue by eventualbuddha** [babel/babel-eslint#429]
Originally posted at eslint/eslint#7854

---

**Tell us about your environment**

macOS Sierra, iTerm 2.0

* **ESLint Version:** 3.12.2
* **Node Version:** 5.1.0
* **npm Version:** 3.3.12

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

```js
{
  "parser": "babel-eslint"
  "rules": {
    "template-curly-spacing": ["error", "never"]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

See demo repo [here](https://github.com/eventualbuddha/eslint-babel-template-curly-spacing-issue).

```js
`a${(function() { return `b${ 0 }c`; })}d`;
```

**What did you expect to happen?**

I expect two errors, one on both sides of the `0` inside the nested template literal.

**What actually happened? Please include the actual, raw output from ESLint.**

I got one error:

```
/Users/donovan/Desktop/eslint-babel-template-curly-issue/index.js
  1:33  error  Unexpected space(s) before '}'  template-curly-spacing

✖ 1 problem (1 error, 0 warnings)
```

eslint is not reporting the space after `${` as a problem. This is due to the difference in tokens between espree and babylon. It may actually be a bug in babylon, but I'm not sure. See the [demo repo](https://github.com/eventualbuddha/eslint-babel-template-curly-spacing-issue) for further analysis.

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
